frontend/src/component/Account/UserSpacesConfig.jsx

```diff
@@ -0,0 +1,228 @@
+import { translate } from 'react-i18next'
+
+import {
+  BtnSwitch,
+  IconButton,
+  ConfirmPopup,
+  ROLE_LIST,
+  TracimComponent,
+  TLM_ENTITY_TYPE as TLM_ET,
+  TLM_CORE_EVENT_TYPE as TLM_CET
+} from 'tracim_frontend_lib'
+
+import { newFlashMessage } from '../../action-creator.sync.js'
+import { deleteWorkspaceMember, getUserWorkspaceList, getWorkspaceMemberList } from '../../action-creator.async.js'
+
+export class UserSpacesConfig extends React.Component {
+  constructor (props) {
+    super()
+
+    this.state = {
+      workspaceList: [],
+      spaceBeingDeleted: null
+    }
+
+    props.registerLiveMessageHandlerList([
+      { entityType: TLM_ET.SHAREDSPACE_MEMBER, coreEntityType: TLM_CET.CREATED, handler: this.handleMemberCreated },
+      { entityType: TLM_ET.SHAREDSPACE_MEMBER, coreEntityType: TLM_CET.MODIFIED, handler: this.handleMemberModified },
+      { entityType: TLM_ET.SHAREDSPACE_MEMBER, coreEntityType: TLM_CET.DELETED, handler: this.handleMemberDeleted }
+    ])
+  }
+
+  handleMemberModified = data => {
+    const { props } = this
+    if (Number(props.userToEditId) !== data.fields.user.user_id) return
+    this.setState(prev => ({
+      workspaceList: prev.workspaceList.map(space =>
+        space.workspace_id === data.fields.workspace.workspace_id
+          ? {
+            ...space,
+            memberList: space.memberList.map(member => member.user_id === Number(props.userToEditId)
+              ? { ...member, do_notify: data.fields.member.do_notify }
+              : member
+            )
+          }
+          : space
+      )
+    }))
+  }
+
+  handleMemberDeleted = data => {
+    const { props } = this
+    if (Number(props.userToEditId) !== data.fields.user.user_id) return
+    this.setState(prev => ({
+      workspaceList: prev.workspaceList.filter(ws =>
+        ws.workspace_id !== data.fields.workspace.workspace_id
+      )
+    }))
+  }
+
+  handleMemberCreated = async data => {
+    const { props } = this
+    if (Number(props.userToEditId) !== data.fields.user.user_id) return
+
+    const newSpace = await this.fillMemberList(data.fields.workspace)
+
+    this.setState(prev => ({
+      workspaceList: [...prev.workspaceList, newSpace].sort(
+        (space1, space2) => space1.workspace_id - space2.workspace_id
+      )
+    }))
+  }
+
+  getWorkspaceList = async () => {
+    const { props } = this
+
+    const fetchGetUserWorkspaceList = await props.dispatch(getUserWorkspaceList(props.userToEditId, false))
+
+    switch (fetchGetUserWorkspaceList.status) {
+      case 200: this.getUserWorkspaceListMemberList(fetchGetUserWorkspaceList.json); break
+      default: props.dispatch(newFlashMessage(props.t('Error while loading user')))
+    }
+  }
+
+  fillMemberList = async space => {
+    const fetchMemberList = await this.props.dispatch(getWorkspaceMemberList(space.workspace_id))
+
+    return {
+      ...space,
+      memberList: fetchMemberList.json || [] // handle error?
```

Comment:
Forgotten comment?

Reply:
nope, this is a comment from a moved code. I decided to leave it here